src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -105,14 +106,53 @@ public static void DuplicateKeysFail()
         [Fact]
         public static void DictionaryOfObject()
         {
-            Dictionary<string, object> obj = JsonSerializer.Parse<Dictionary<string, object>>(@"{""Key1"":1}");
-            Assert.Equal(1, obj.Count);
-            JsonElement element = (JsonElement)obj["Key1"];
-            Assert.Equal(JsonValueType.Number, element.Type);
-            Assert.Equal(1, element.GetInt32());
+            {
+                Dictionary<string, object> obj = JsonSerializer.Parse<Dictionary<string, object>>(@"{""Key1"":1}");
+                Assert.Equal(1, obj.Count);
+                JsonElement element = (JsonElement)obj["Key1"];
+                Assert.Equal(JsonValueType.Number, element.Type);
+                Assert.Equal(1, element.GetInt32());
 
-            string json = JsonSerializer.ToString(obj);
-            Assert.Equal(@"{""Key1"":1}", json);
+                string json = JsonSerializer.ToString(obj);
+                Assert.Equal(@"{""Key1"":1}", json);
+            }
+
+            {
+                IDictionary<string, object> obj = JsonSerializer.Parse<IDictionary<string, object>>(@"{""Key1"":1}");
+                Assert.Equal(1, obj.Count);
+                JsonElement element = (JsonElement)obj["Key1"];
+                Assert.Equal(JsonValueType.Number, element.Type);
+                Assert.Equal(1, element.GetInt32());
+
+                string json = JsonSerializer.ToString(obj);
+                Assert.Equal(@"{""Key1"":1}", json);
+            }
+        }
+
+        [Fact]
+        public static void ImplementsIDictionaryOfObject()
+        {
+            var input = new StringToObjectIDictionaryWrapper(new Dictionary<string, object>
+            {
+                { "Name", "David" },
+                { "Age", 32 }
```

Comment:
We need to leave this age here forever.

Reply:
Nah, we'll update this with every release.